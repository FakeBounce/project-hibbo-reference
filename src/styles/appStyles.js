/* eslint global-require: 0 */
import { Dimensions } from 'react-native';

import { iconVideos } from 'utils/storedData';

const { width } = Dimensions.get('window');

export default {
  prefix: {
    fixedIcon: 'icon_opac/',
  },
  event: {
    none: 0,
    gateau: 1,
    sapin: 2,
    oeuf: 3,
    souris: 4,
    bonbon: 11,
    manette: 12,
    moto: 13,
    voiture: 14,
    cadeau_vert: 15,
    cadeau_bleu: 16,
    courses: 21,
    rangement: 22,
    crayon: 23,
    marteau: 24,
    lapin: 31,
    chat: 32,
    chien: 33,
    heureux: 41,
    betise: 42,
    system: {
      reset: 0xffff,
      setLocal: 0xfff0,
      setVolume: 0xfff1,
      setObjectif: 0xfff2,
    },
  },
  icons: iconVideos,
  maxTextSize: 70,
  colors: {
    white: '#FFF',
    black: '#000',
    darkGrey: '#303030',
    darkBlue: '#06263D',
    darkBlueModal: '#06263D',
    grey: '#979797',
    greyOpac: 'rgba(151, 151, 151, 0.18)',
    whiteOpac: 'rgba(255, 255, 255, 0.30)',
    green: '#16EBC5',
    fb: '#4100C0',
    google: '#D5492F',
    primary: '#413D3D',
    textGrey: '#A7A7A7',
    buttonGrey: '#EFEFEF',
    lightGrey: '#9B9B9B',
    lightGreyBullet: '#E5E3DD',
    blueGreen: '#16EBC5',
    borderGrey: 'rgba(74, 74, 74, 0.23)',
    blackModal: 'rgba(0, 0, 0, 0.5)',
    toggleButtonGrey: '#4A4A4A',
    yellow: '#FFDF2D',
    moneyPotText: '#FFFE82',
    buttonYellow: '#FFFE82',
    cardBlue: '#697A99',
    gold: '#FEE9BF',
    red: '#FC5151',
    error: '#FF6969',
    textLink: '#687A99',
    messageGrey: '#DDE6EC',
    blue: '#0084ff',
    balanceGreen: '#3CDD80',
    pelicanColor: '#A685FF',
    deepBlue: '#028DAF',
    flatPink: '#FF6682',
    flatOrange: '#FFB434',
    flatPurple: '#82438F',
    white2pac: 'rgba(255,255,255,0.2)',
    white1pac: 'rgba(255,255,255,0.1)',
    placeholderGrey: 'rgba(229, 227, 221, 0.5)',
    greenStrong: '#50E3C2',
    orangeNotStrong: '#F5A623',
    inputLabel: '#C3C3C3',
    inputGrey: '#E8E8E8',
  },
  transaction: {
    minimunForAnimation: 1000,
  },
  time: {
    moneyTransaction: 2500,
  },
  textSizes: {
    xxxsmall: 10,
    xxsmall: 13,
    xsmall: 16,
    small: 18,
    regular: 22,
    normal: 24,
    xnormal: 26,
    big: 36,
    xbig: 50,
    xxbig: 58,
  },
  size: {
    cardMargin: 10,
    maxBubbleWidth: 190,
  },
  cardStyle: {
    shadowOpacity: 0,
    shadowColor: 'transparent',
    borderRadius: 10,
  },
  fonts: {
    circularBook: 'Circular-Std-Book',
    circularBookItalic: 'Circular-Std-Book-Italic',
    circularMedium: 'Circular-Std-Medium',
    circularMediumItalic: 'Circular-Std-Medium-Italic',
    circularBold: 'Circular-Std-Bold',
    circularBoldItalic: 'Circular-Std-Bold-Italic',
    circularBlackItalic: 'Circular-Std-Black-Italic',
    pixel: 'LCDDot-TR',
    chewy: 'Chewy',
    peepo: 'Peepo',
  },
  get colorList() {
    return [
      this.colors.deepBlue,
      this.colors.flatPink,
      this.colors.flatOrange,
      this.colors.flatPurple,
    ];
  },
  colorListSuffix: ['_bleu', '_rose', '_orange', '_violet'],
  swiperRange: [0, width / 2, width, width * 1.5, width * 2],
};
