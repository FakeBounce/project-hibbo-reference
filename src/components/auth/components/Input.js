import React, { PureComponent } from 'react';
import {
  string,
  func,
  shape,
  bool,
  oneOfType,
  object,
  number,
  array,
} from 'prop-types';
import { getTranslations } from 'utils/i18n';
import appStyles from 'styles/appStyles';

import { SContainerInput, SInput } from '../styles';

class Input extends PureComponent {
  constructor(props) {
    super(props);
    this.inputText = null;
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.label !== nextProps.label) {
      this.inputText.focus();
    }
  }

  getKeyboardType = label => {
    switch (label) {
      case getTranslations('auth.label.email'):
        return 'email-address';
      case getTranslations('auth.label.phone'):
        return 'phone-pad';
      default:
        return 'default';
    }
  };

  getReturnKeyType = label => {
    if (label === getTranslations('auth.label.confirmpassword')) return 'done';
    return 'next';
  };

  getAutoCapitalize = label => {
    if (
      label === getTranslations('auth.label.firstname') ||
      label === getTranslations('auth.label.lastname')
    )
      return 'sentences';
    return 'none';
  };

  getResChange(text, keyboardType) {
    let res = text;
    if (keyboardType !== 'default') {
      res = text.replace(/\s/g, '');
    }
    if (keyboardType === 'phone-pad') {
      res = res.replace(/[-|.|(|)]/g, '');
    }
    return res;
  }

  needSecurity(label) {
    return (
      label === getTranslations('auth.label.password') ||
      label === getTranslations('auth.label.confirmpassword')
    );
  }

  render() {
    const {
      form,
      label,
      name,
      onChange,
      validate,
      error,
      title,
      hide,
      autoFocus,
      inputTextSize,
      inputStyle,
    } = this.props;
    const keyboardType = this.getKeyboardType(label);

    return (
      <SContainerInput>
        <SInput
          key={keyboardType}
          innerRef={ref => {
            this.inputText = ref;
          }}
          value={form[name]}
          autoCapitalize={this.getAutoCapitalize(label)}
          autoCorrect={false}
          autoFocus={autoFocus}
          keyboardType={keyboardType}
          onChangeText={text => {
            onChange(name, this.getResChange(text, keyboardType));
          }}
          onSubmitEditing={() => validate(title)}
          placeholder={label}
          placeholderTextColor={
            error ? appStyles.red : appStyles.toggleButtonGrey
          }
          returnKeyType={this.getReturnKeyType(label)}
          secureTextEntry={this.needSecurity(label) && hide}
          inputTextSize={inputTextSize}
          style={inputStyle}
          underlineColorAndroid="transparent"
        />
      </SContainerInput>
    );
  }
}

Input.defaultProps = {
  hide: false,
  inputStyle: {
    height: 75,
    marginTop: 90,
  },
  autoFocus: true,
};

Input.propTypes = {
  inputStyle: oneOfType([object, number, array]),
  hide: bool,
  autoFocus: bool,
  name: string.isRequired,
  label: string.isRequired,
  onChange: func.isRequired,
  validate: func.isRequired,
  form: shape({
    email: string,
    firstName: string,
    lastName: string,
    pwd: string,
  }).isRequired,
  error: bool.isRequired,
  title: string.isRequired,
  inputTextSize: number.isRequired,
};

export default Input;
