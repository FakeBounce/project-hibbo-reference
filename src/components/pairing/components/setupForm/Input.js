import React from 'react';
import { Animated, TextInput } from 'react-native';
import {
  string,
  func,
  bool,
  object,
  array,
  number,
  oneOfType,
  shape,
  instanceOf,
} from 'prop-types';
import styled from 'styled-components/native';

import appStyles from 'styles/appStyles';

const SBorder = styled(Animated.View)`
  border-bottom-width: 2px;
  border-bottom-color: ${props => props.theme.colors.white};
  margin-horizontal: 20px;
  overflow: hidden;
`;
const SInput = styled(Animated.createAnimatedComponent(TextInput))`
  height: 45px;
  padding: 5px;
  font-size: ${props => props.theme.textSizes.regular};
  text-align: center;
  font-family: ${props => props.theme.fonts.circularBold};
`;

const Input = ({
  name,
  defaultValue,
  placeholder,
  setData,
  validate,
  error,
  secureEntry,
  keyboardType,
  capitalize,
  style,
  returnKeyType,
  maxLength,
  innerRef,
  animations,
  autoFocus,
}) => {
  return (
    <SBorder
      style={[
        style,
        {
          opacity: animations.borderOpacity,
          transform: [{ translateY: animations.borderTranslateY }],
        },
      ]}
    >
      <SInput
        key={keyboardType}
        value={defaultValue}
        autoCapitalize={capitalize}
        autoCorrect={false}
        autoFocus={autoFocus}
        keyboardType={keyboardType}
        onChangeText={text => {
          setData(name, text);
        }}
        innerRef={innerRef}
        onSubmitEditing={() => {
          validate(name);
        }}
        placeholder={placeholder}
        placeholderTextColor={
          error ? appStyles.colors.google : appStyles.colors.white
        }
        returnKeyType={returnKeyType}
        style={[
          style,
          {
            color: error ? appStyles.colors.google : appStyles.colors.white,
            opacity: animations.inputOpacity,
            transform: [{ translateY: animations.inputTranslateY }],
          },
        ]}
        secureTextEntry={secureEntry}
        underlineColorAndroid="transparent"
        maxLength={maxLength}
      />
    </SBorder>
  );
};

Input.defaultProps = {
  capitalize: 'words',
  style: {},
  returnKeyType: 'send',
  secureEntry: false,
  keyboardType: 'default',
  maxLength: 100,
  innerRef: null,
  autoFocus: false,
};

Input.propTypes = {
  innerRef: func,
  maxLength: number,
  capitalize: string,
  name: string.isRequired,
  defaultValue: string.isRequired,
  placeholder: string.isRequired,
  setData: func.isRequired,
  validate: func.isRequired,
  error: bool.isRequired,
  secureEntry: bool,
  keyboardType: string,
  style: oneOfType([object, number, array]),
  returnKeyType: string,
  animations: shape({
    borderOpacity: instanceOf(Animated.Value).isRequired,
    borderTranslateY: instanceOf(Animated.Value).isRequired,
    inputOpacity: instanceOf(Animated.Value).isRequired,
    inputTranslateY: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  autoFocus: bool,
};

export default Input;
