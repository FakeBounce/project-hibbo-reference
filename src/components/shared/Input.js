import React from 'react';
import { Dimensions } from 'react-native';
import { string, func, bool, oneOfType, object, number } from 'prop-types';

import theme from 'styles/appStyles';

import * as Styles from './styles';

const { width } = Dimensions.get('window');
const Input = ({
  name,
  keyboardType,
  capitalize,
  onChange,
  innerRef,
  onSubmit,
  placeholder,
  error,
  returnKeyType,
  style,
  secureEntry,
  maxLength,
  centered,
  value,
  inputWidth,
}) => {
  return (
    <Styles.SInputContainer centered={centered}>
      <Styles.SInput
        key={keyboardType}
        value={value}
        autoCapitalize={capitalize}
        autoCorrect={false}
        keyboardType={keyboardType}
        onChangeText={text => {
          onChange(name, text);
        }}
        innerRef={innerRef}
        onSubmitEditing={() => {
          onSubmit(name);
        }}
        placeholder={placeholder}
        placeholderTextColor={
          error ? theme.colors.google : theme.colors.inputGrey
        }
        returnKeyType={returnKeyType}
        style={style}
        inputWidth={inputWidth}
        secureTextEntry={secureEntry}
        underlineColorAndroid="transparent"
        maxLength={maxLength}
      />
      <Styles.SBorderInput inputWidth={inputWidth} />
    </Styles.SInputContainer>
  );
};

Input.defaultProps = {
  keyboardType: 'default',
  capitalize: 'words',
  innerRef: null,
  returnKeyType: 'next',
  style: {},
  secureEntry: false,
  maxLength: 500,
  centered: false,
  inputWidth: width,
  onSubmit: () => {},
};

Input.propTypes = {
  name: string.isRequired,
  keyboardType: string,
  capitalize: string,
  onChange: func.isRequired,
  innerRef: func,
  onSubmit: func,
  placeholder: string.isRequired,
  error: bool.isRequired,
  returnKeyType: string,
  style: oneOfType([string, number, object]),
  secureEntry: bool,
  maxLength: number,
  centered: bool,
  value: oneOfType([string, number]).isRequired,
  inputWidth: number,
};

export default Input;
