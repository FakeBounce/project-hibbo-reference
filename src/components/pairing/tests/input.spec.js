import {
  TextInput,
} from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import appStyles from 'styles/appStyles';

import Input from '../components/setupForm/Input';

describe('Input', () => {
  const fn1 = jest.fn();
  const fn2 = jest.fn();
  const input = (
    <Input
      value=""
      label=""
      name=""
      defaultValue=""
      placeholder=""
      validate={fn2}
      setData={fn1}
      error={false}
    />
  );
  const component = renderer.create(input).toJSON();
  let tree = shallow(input);

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should handle text change', () => {
    tree.find(TextInput).first().simulate('ChangeText');
    expect(fn1).toHaveBeenCalled();
  });

  it('should handle submit editing', () => {
    tree.find(TextInput).first().simulate('SubmitEditing');
    expect(fn2).toHaveBeenCalled();
  });

  it('should have a red placeholder if error', () => {
    tree = shallow(
      <Input
        value=""
        label=""
        name=""
        defaultValue=""
        placeholder=""
        validate={fn2}
        setData={fn1}
        error
      />,
    );

    expect(tree.find(TextInput).get(0).props.placeholderTextColor).toEqual(appStyles.colors.google);
    expect(tree.find(TextInput).get(0).props.style).toEqual([
      undefined,
      {},
      {
        color: appStyles.colors.google,
      },
    ]);
  });
});
