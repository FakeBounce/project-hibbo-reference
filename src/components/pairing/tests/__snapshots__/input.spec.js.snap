// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render the component 1`] = `
<View
  style={
    Array [
      undefined,
      Object {},
    ]
  }
>
  <TextInput
    autoCapitalize="words"
    autoCorrect={false}
    autoFocus={false}
    keyboardType="default"
    maxLength={100}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder=""
    placeholderTextColor="#FFDF2D"
    returnKeyType="send"
    secureTextEntry={false}
    style={
      Array [
        undefined,
        Object {},
        Object {
          "color": "#FFDF2D",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
